oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -3039,3 +3039,255 @@ def sort(self, ascending=True):
+        1    1.5
+        2    2.0
+        3    3.0
+        4    4.0
+        dtype: float64
+
+        A rolling mean with a size of 3, centered around the current:
+        >>> sa.rolling_mean(-1,1)
+        dtype: float
+        Rows: 5
+        [None, 2.0, 3.0, 4.0, None]
+
+        Pandas equivalent:
+        >>> pandas.rolling_mean(series, 3, center=True)
+        0   NaN
+        1     2
+        2     3
+        3     4
+        4   NaN
+        dtype: float64
+
+        A rolling mean with a window including the current and the 2 entries
+        following:
+        >>> sa.rolling_mean(0,2)
+        dtype: float
+        Rows: 5
+        [2.0, 3.0, 4.0, None, None]
+
+        A rolling mean with a window including the previous 2 entries NOT
+        including the current:
+        >>> sa.rolling_mean(-2,-1)
+        dtype: float
+        Rows: 5
+        [None, None, 1.5, 2.5, 3.5]
+        """
+        if min_observations is None:
+            min_observations = (1 << 64) - 1
+        if min_observations < 0:
+            raise ValueError("min_observations must be a positive integer")
+        agg_op = None
+        if self.dtype() is array.array:
+            agg_op = '__builtin__vector__avg__'
+        else:
+            agg_op = '__builtin__avg__'
+        return SArray(_proxy=self.__proxy__.rolling_apply(agg_op, window_start, window_end, min_observations))
+
+    def rolling_sum(self, window_start, window_end, min_observations=None):
+        """
+        Calculate a new SArray of the sum of different subsets over this
+        SArray.
+
+        Also known as a "moving sum" or "running sum". The subset that
+        the sum is calculated over is defined as an inclusive range relative
+        to the position to each value in the SArray, using `window_start` and
+        `window_end`. For a better understanding of this, see the examples
+        below.
+
+        Parameters
+        ----------
+        window_start : int
+            The start of the subset to calculate the sum relative to the
+            current value.
+
+        window_end : int
+            The end of the subset to calculate the sum relative to the current
+            value. Must be greater than `window_start`.
+        
+        min_observations : int 
+            Minimum number of non-missing observations in window required to
+            calculate the sum (otherwise result is None). None signifies that
+            the entire window must not include a missing value. A negative
+            number throws an error.
+
+        Returns
+        -------
+        out : SArray
+
+        Examples
+        --------
+        >>> import pandas
+        >>> sa = SArray([1,2,3,4,5]) 
+        >>> series = pandas.Series([1,2,3,4,5])
+
+        A rolling sum with a window including the previous 2 entries including
+        the current:
+        >>> sa.rolling_sum(-2,0)
+        dtype: int 
+        Rows: 5
+        [None, None, 6, 9, 12]
+
+        Pandas equivalent:
```

Comment:
I think this belongs more in the translator. I would personally prefer removing them.